src/main/java/edu/cornell/kfs/coa/batch/service/impl/CreateClosedAccountsCsvServiceImpl.java

```diff
@@ -0,0 +1,306 @@
+                    + " and a TO_DATE = " + fromToRange.get(CuCoaBatchConstants.ClosedAccountsFileCreationConstants.TO_DATE)); 
+            } else {
+                LOG.info("getClosedAccountsDateRange: Periodic file will NOT be created.");
+            }
+            
+        } else {
+            LOG.error("getClosedAccountsDateRange: Unknown Type of Closed Account File to Create = " + typeOfClosedAccountFileToCreate.get(CuCoaBatchConstants.ClosedAccountsFileCreationConstants.FILE_DATA_CONTENT_TYPE_IS));
+        }
+        return fromToRange;
+    }
+    
+    private Date getJavaSqlDateForComputedDate(int daysPrevious) {
+        LocalDate computedFromDate = computeDateBeingDaysPreviousFromToday(daysPrevious);
+        java.sql.Date sqlDate;
+        try {
+            sqlDate = getDateTimeService().convertToSqlDate(computedFromDate.toString());
+        } catch (ParseException pe) {
+            LOG.error("getJavaSqlDateForComputedDate: ParseException caught for computedFromDate = " + computedFromDate.toString());
+            sqlDate = null;
+        }
+        return sqlDate;
+    }
+    
+    private LocalDate computeDateBeingDaysPreviousFromToday(int daysPrevious) {
+        LOG.info("computeDateBeingDaysPreviousFromToday: daysPrevious = " + daysPrevious);
+        DateTime todayAsDateTime = new DateTime(getDateTimeService().getCurrentDate());
+        DateTime datePreviousAsDateTime = todayAsDateTime.minusDays(daysPrevious);
+        LOG.info("computeDateBeingDaysPreviousFromToday: datePreviousAsDateTime = " + datePreviousAsDateTime.toLocalDate().toString());
+        return datePreviousAsDateTime.toLocalDate();
+    }
+    
+    private void writeClosedAccountsToCsvFormattedFile(Map<String, Date> fileDataRangeSpecified) throws IOException {
+        LOG.info("writeClosedAccountsToCsvFormattedFile: Prior to ensuring directories to hold outbound file exist.");
+        ensureOutbundDirectoriesExist();
+        
+        LOG.info("writeClosedAccountsToCsvFormattedFile: Prior to obtaining data with SQL query for date range. This may take a while.");
+        List<ClosedAccount> closedAccountsData = getClosedAccountsByDateRangeDao().obtainClosedAccountsDataFor(fileDataRangeSpecified);
+        
+        if (ObjectUtils.isNotNull(closedAccountsData) && closedAccountsData.size() > 0) {
+            LOG.info("writeClosedAccountsToCsvFormattedFile: Have the data. Will now write out " + closedAccountsData.size() + " closed accounts to CSV file.");
+            writeClosedAccountsToCsvFile(closedAccountsData);
+        } else {
+            LOG.info("writeClosedAccountsToCsvFormattedFile: NO CSV file being created. " 
+                    + (ObjectUtils.isNotNull(closedAccountsData) ? (closedAccountsData.size() + " accounts being written to file.") : ("null returned for requested account data.") ));
+        }
+    }
+    
+    private void ensureOutbundDirectoriesExist() {
```

Comment:
I spoke with Nancy about this, and spring should create the directories at boot time if the directories are defined in <property name="directoryPath"> similar to below.  

<property name="batchFileDirectories">
            <list merge="true">
                <value>${staging.directory}/coa/whatever</value>
            </list>
        </property>

Reply:
Jay and I discussed this code review suggestion again when it would now work as initially described.  Based on that second information sharing, it was concluded that the existing code is the solution to implement.